qiskit_experiments/characterization/spectroscopy.py

```diff
@@ -0,0 +1,408 @@
+        )
+
+        if plot and plotting.HAS_MATPLOTLIB:
+            ax = plotting.plot_curve_fit(fit_fun, best_fit, ax=ax)
+            ax = plotting.plot_scatter(xdata, ydata, ax=ax)
+            self._format_plot(ax, best_fit)
+            figures = [ax.get_figure()]
+        else:
+            figures = None
+
+        return best_fit, figures
+
+    @staticmethod
+    def _fit_quality(
+        fit_amp: float,
+        fit_sigma: float,
+        fit_freq: float,
+        fit_offset: float,
+        reduced_chisq: float,
+        xdata: np.array,
+        ydata: np.array,
+        fit_sigma_err: Optional[float] = None,
+    ) -> str:
+        """Algorithmic criteria for whether the fit is good or bad.
+
+        A good fit has:
+            - a reduced chi-squared less than 3,
+            - a peak within the scanned frequency range,
+            - a standard deviation that is not larger than the scanned frequency range,
+            - a standard deviation that is wider than the smallest frequency increment,
+            - a signal-to-noise ratio, defined as the amplitude of the peak divided by the
+              square root of the median y-value less the fit offset, greater than a
+              threshold of two, and
+            - a standard error on the sigma of the Gaussian that is smaller than the sigma.
+
+        Args:
+            fit_amp: Amplitude of the fitted peak.
+            fit_sigma: Standard deviation of the fitted Gaussian.
+            fit_freq: Frequency of the fitted peak.
+            fit_offset: Offset of the fit.
+            reduced_chisq: Reduced chi-squared of the fit.
+            xdata: x-values, i.e. the frequencies.
+            ydata: y-values, i.e. the measured signal.
+            fit_sigma_err: Errors on the standard deviation of the fit.
+
+        Returns:
+            computer_bad or computer_good if the fit passes or fails the criteria, respectively.
+        """
+        min_freq = xdata[0]
+        max_freq = xdata[-1]
+        freq_increment = xdata[1] - xdata[0]
+
+        snr = abs(fit_amp) / np.sqrt(abs(np.median(ydata) - fit_offset))
+        fit_width_ratio = fit_sigma / (max_freq - min_freq)
+
+        # pylint: disable=too-many-boolean-expressions
+        if (
+            min_freq <= fit_freq <= max_freq
+            and 1.5 * freq_increment < fit_sigma
+            and fit_width_ratio < 0.25
+            and reduced_chisq < 3
+            and (fit_sigma_err is None or (fit_sigma_err < fit_sigma))
+            and snr > 2
+        ):
+            return "computer_good"
+        else:
+            return "computer_bad"
+
+    @classmethod
+    def _format_plot(cls, ax, analysis_result):
+        """Format curve fit plot."""
+        ax.tick_params(labelsize=14)
+        ax.set_xlabel(f"Frequency ({analysis_result['unit']})", fontsize=16)
+        ax.set_ylabel("Signal [arb. unit.]", fontsize=16)
+        ax.grid(True)
+
+
+class Spectroscopy(BaseExperiment):
```

Comment:
Perhaps this is too broad naming. We may want to implement spectroscopy for resonator and this causes naming conflict. What about `QubitSpectroscopy` of `QubitFgeSpectroscopy` (i.e. `Fef` for 1-2 transition)?

Reply:
Done here: 3f43c74